src/Compilers/CSharp/Portable/Symbols/MethodSymbolExtensions.cs

```diff
@@ -225,5 +225,28 @@ internal static bool IsValidUnscopedRefAttributeTarget(this MethodSymbol method)
                 !method.IsConstructor() &&
                 !method.IsInitOnly;
         }
+
+#nullable enable
+        internal static ImmutableArray<DeclarationScope> GetByValueParameterDeclarationScopes(this MethodSymbol? method)
```

Comment:
>GetByValueParameterDeclarationScopes [](http://example.com/codeflow?start=57&length=36)

I think it would be helpful if these methods are named as "...DeclaredScopes" or "...EffectiveScopes". Currently, it's not clear if "...DeclarationScopes" refers to the enum type name or the declared values.